client/src/components/Related/RelatedProducts.jsx

```diff
@@ -1,20 +1,50 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState } from 'react';
 import ProductCard from './ProductCard.jsx';
 
+
+
 const RelatedProducts = (props) => {
+  const [position, setPosition] = useState(0);
+  const componentName = 'RelatedProducts';
+
   return (
-    <div className='related-products-container'>
-      {props.relatedProducts ?
+    <>
+      <div className='related-products-container' style={{marginLeft: `-${position}px`}}>
+        {props.relatedProducts ?
 
 
-      props.relatedProducts.map((product) => {
-        return <ProductCard key={product.id} product={product} generateStars={props.generateStars}/>
-      })
-      : null
-      }
-    </div>
+          props.relatedProducts.map((product) => {
+            return <ProductCard
+              key={product.id}
+              product={product}
+              generateStars={props.generateStars}
+              isFetching={props.isFetching}
+              setIsFetching={props.setIsFetching}
+              parentComponent={componentName}
```

Comment:
I see you pass down the parent component name to ProductCard and you use that in ProductCard to conditionally render styles it looks like, but I don't understand exactly what's going on. Nothing to change, just curious to hear lol